src/nodes.coffee

```diff
@@ -4518,6 +4526,36 @@ exports.StringWithInterpolations = class StringWithInterpolations extends Base
     call = element.unwrapAll?()
     @csx and call instanceof CSXElement
 
+  astType: -> 'TemplateLiteral'
+
+  astProperties: (o) ->
+    elements = @extractElements o
+    [..., last] = elements
+
+    quasis = []
+    expressions = []
+
+    for element, index in elements
+      if element instanceof StringLiteral
+        quasis.push new TemplateElement(
+          element.originalValue
+          tail: element is last
+        ).withLocationDataFrom(element).ast o
+      else
+        expressions.push element.unwrap().ast o
+
+    {expressions, quasis, @quote}
+
+exports.TemplateElement = class TemplateElement extends Base
+  constructor: (@value, {@tail} = {}) ->
+    super()
+
+  astProperties: ->
+    return
+      value:
+        raw: @value
```

Comment:
There is also a `value.cooked` property in Babel-style `TemplateElement` AST, but I'm not sure the difference between `raw` and `cooked` so omitted at least for now. I'm guessing it's related to eg escapes and that's something that deserves its own investigation/PR